server/sonar-web/src/main/js/apps/sessions/components/LoginFormContainer.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import LoginForm from './LoginForm';
+import { doLogin } from '../../../store/rootActions';
+import { fetchAppState } from '../../../store/rootActions';
+import { IdentityProvider, getIdentityProviders } from '../../../api/users';
+import { getBaseUrl } from '../../../helpers/urls';
+
+interface Props {
+  doLogin: (login: string, password: string) => Promise<void>;
+  fetchAppState: () => Promise<any>;
+  location: { hash?: string; pathName: string; query: { return_to?: string } };
+}
+
+interface State {
+  identityProviders?: IdentityProvider[];
+  onSonarCloud: boolean;
+}
+
+class LoginFormContainer extends React.PureComponent<Props, State> {
+  mounted: boolean;
+  state: State = { onSonarCloud: false };
+
+  static contextTypes = {
+    router: PropTypes.object
+  };
+
+  componentDidMount() {
+    this.mounted = true;
+    Promise.all([
+      getIdentityProviders(),
+      this.props.fetchAppState()
+    ]).then(([identityProvidersResponse, appState]) => {
+      if (this.mounted) {
+        this.setState({
+          onSonarCloud:
+            appState.settings != undefined &&
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') != should be !== [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1440)